docs/jsonapi-spec.md

```diff
@@ -591,6 +628,102 @@ See [Multi-Document Failure Considerations](#multi-document-failure-consideratio
 
 If an error occurs the command will not return `data`.
 
+
+### findCollections Command
+
+`findCollections` returns all collections from a given namespace.
+
+There is no payload. The `namespace` is given as `{{base_url}}{{json_port}}/v1/{namespace}`.
+
+*Syntax:*
+
+```bnf
+<findCollections-command>            ::= findCollections
+<findCollections-command-response>   ::= status.collections: ["col1", "col2"]
+```
+
+#### findCollections Command Options
+
+None. 
+
+#### findCollections Command Response
+
+| Response Elements | Description                                                                        |
+| ----------------- | ---------------------------------------------------------------------------------- |
+| `status`          | Status has `collections` field with array of the available collection names.       |
+| `errors`          | If the provided namespace does not exist, return `NAMESPACE_DOES_NOT_EXIST`.       |
+
+
+### findOneAndReplace Command
+
+`findOneAndReplace` replaces the first document in the collection that matches the filter. Optionally use a `sort-clause` to determine which document is modified.
+
+*Syntax:*
+
+```bnf
+<find-one-and-replace-command> ::= findOneAndReplace <find-one-and-replace-command-payload> 
+<find-one-and-replace-command-payload> ::= <filter-clause>? 
+                             <sort-clause>? 
+                             <replacement>?
+                             <find-one-and-replace-command-options>?
+
+<find-one-and-replace-command-option> ::= (<returnDocument>)*
+```
+
+*Sample:*
+
+```json5
+// Replaces a single document based on the specified filter and sort. Returns a document in response.
+{ "findOneAndReplace" : 
+    { "filter" : {<filter-clause>}, 
+      "sort" : {<sort-clause>}, 
+      "replacement" : {<document-to-replace>}, 
+      "options" : 
+          {"returnDocument" : "before/after"} 
+     }
+}
+```
+
+#### findOneAndReplace Command Order of Operations
+
+`findOneAndReplace` commands are processed using the following order of operation:
+
+1.  `<filter-clause>` is applied to the collection to select a single candidate document. Example:  `"filter": {"location": "London"}`.
```

Comment:
Since sort will be applied, it will potentially select multiple candidates.

Reply:
Thanks, updated in latest commit.